scripts/fetch.mjs

```
@@ -0,0 +1,25 @@
+import { getAccessToken } from "./accessToken.mjs";
+
+async function doFetch(url, customOptions = {}, shouldUseAuth = false) {
+    try {
+      const fetchOptions = {
+        headers: {
+          "Content-Type": "application/json",
+        },
+        ...customOptions,
+      };
+  
+      if (shouldUseAuth) {
+        const accessToken = getAccessToken();
+        fetchOptions.headers.Authorization = `Bearer ${accessToken}`;
+      }
+      const response = await fetch(url, fetchOptions);
+      const json = await response.json();
+      return json;
+    } catch (error) {
+      throw new Error(error);
```

`catch block`: to suppress an error
`throw`: to create an error

Understand the differences, they have different purposes. Here you're basically catching an error and throwing it again.

**Suggestion:** remove the throw from the catch block and use the try/catch block in a UI/UX code block e.g event listener.